### PR TITLE
[MM-67797] Expose ability to update the title template for popouts

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
Popout windows currently receive a `titleTemplate` only at creation time via `openPopout`. Once a popout is open, there's no way for the web app to change how the window title is formatted — for example, to update the template when the user navigates to a different context within the same popout.

This PR adds an `updatePopoutTitleTemplate` method to the Desktop API. When called from a popout view, it updates the stored title template and immediately re-renders the window title using the existing title components (channel name, team name, server name).

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-67797

#### Release Note
```release-note
NONE
```

## Change Impact: 🟡 Medium

**Reasoning:** The change introduces a new public API method and IPC channel and updates view management to support dynamic popout title templates; these are additive but span multiple modules (preload API, IPC, popout manager, view manager, and types), increasing surface area.

**Regression Risk:** Moderate. Although behavior is additive and unit tests were added, the change touches shared communication channels and view lifecycle code that could affect UI title updates or event propagation if edge cases are missed.

**QA Recommendation:** Perform focused manual QA: smoke-test popout creation and normal title rendering, verify calling updatePopoutTitleTemplate from a popout updates the title immediately across scenarios (popout vs tab, missing view props), and run related automated tests. Skipping manual QA is not recommended due to cross-module impact.

*Generated by CodeRabbitAI*

### CODE REVIEW

File: src/app/windows/popoutManager.ts [hunk 1]
```diff
@@ -392,6 +392,14 @@ export class PopoutManager {
         view.sendToRenderer(MESSAGE_FROM_PARENT, channel, ...args);
     };
 
+    private handleUpdatePopoutTitleTemplate = (event: IpcMainEvent, titleTemplate: string) => {
+        const webContentsView = WebContentsManager.getViewByWebContentsId(event.sender.id);
+        if (!webContentsView) {
+            return;
+        }
+        ViewManager.updateViewTitleTemplate(webContentsView.id, titleTemplate);
+    };
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply src/app/windows/popoutManager.ts:407]: @coderabbit This is resolved, and so is the other comment around the validation, since it's not necessary. Please re-review.

File: src/app/windows/popoutManager.ts [hunk 2]
```diff
@@ -392,6 +392,20 @@ export class PopoutManager {
         view.sendToRenderer(MESSAGE_FROM_PARENT, channel, ...args);
     };
 
+    private handleUpdatePopoutTitleTemplate = (event: IpcMainEvent, titleTemplate: string) => {
+        if (typeof titleTemplate !== 'string') {
+            return;
+        }
```
Comment (Willyfrog) [inline src/app/windows/popoutManager.ts:398]: nitpick: why wouldn't titleTemplate be a string if the function signature requires it being a string?

File: src/common/views/viewManager.ts [hunk 3]
```diff
@@ -132,6 +132,20 @@ export class ViewManager extends EventEmitter {
         this.emit(VIEW_TITLE_UPDATED, view.id);
     };
 
+    updateViewTitleTemplate = (viewId: string, titleTemplate: string) => {
+        log.debug('updateViewTitleTemplate', {viewId});
+
+        const view = this.views.get(viewId);
```
Comment (Willyfrog) [inline src/common/views/viewManager.ts:138]: should this check if the view is actually a window? as in the other file?

File: src/app/windows/popoutManager.ts (hunk 2 above)
Comment (devinbinnie) [reply src/app/windows/popoutManager.ts:398]: There's no enforcement for the client to send up something that conforms to the type, this is a security hardening measure to make sure that we don't consume a `number` and it causes a `TypeError` when we try to run `replace` on it.

File: src/common/views/viewManager.ts (hunk 3 above)
Comment (devinbinnie) [reply src/common/views/viewManager.ts:138]: Regular views make use of `titleTemplate` so I think we're okay here. Worst case it would just update the title in the same way, which isn't a huge deal.
